### ISSUE TITLE
Ability to overwrite logical operations (`&&`, `||`)

### ISSUE THREAD

**Issue by tp** [cel-expr/cel-go#461]
Not sure if this is a known issue/lack of feature, of a bug. I expect the former, but if it's the latter I can write an isolated program showing the issue.

**Issue**

For my custom type I was able to overwrite e.g. the `Adder` using `_+_`. Also when I use `+` with types that don't have the overwrite, I am getting a type check error like

```
type-check error: ERROR: <input>:1:56: found no matching overload for '_+_' applied to '(Predicate, Predicate)'
```

**Example**

I declared a new type `Predicate` and wanted to overwrite `&&` like

```
			cel.Declarations(
				decls.NewFunction("_&&_",
					decls.NewOverload("query_predicate_and",
						[]*exprpb.Type{PredicateType, PredicateType}, PredicateType),
				),
			),
```

The type check passes (unlike in the `+` overwrite example above), but at runtime I am getting a `no such overload` error.

Is there a reason / settings that `&&`/`||` is not overwrite-able?

**Comment by TristonianJones** [comment]
It's strongly discouraged to overload operators.

The logical operators are not designed to be overwritten as they would affect evaluation semantics in ways that might break the language guarantees.

However, add should work. It sounds like you have two issues though. One with getting the overload to be found by the type-checker and the other with the runtime resolution. Implementing the `Adder` will solve the runtime binding. Type-checking will probably require an `AbstractType` (fixed in #460) or protobuf message type for your `Predicate`. You will also need to copy of all the supported overload signatures that you want from the standard `checker/standard.go` set (literally a copy) with your customizations woven in to a `cel.Declarations` option. You will need to use a `cel.NewCustomEnv` to make this work.

**Comment by tp** [comment]
> However, add should work. 

It does (for my custom `decimal` type). Even without the `traits.Adder`. Just implementing `_+_` was sufficient.

> It's strongly discouraged to overload operators.
>
> The logical operators are not designed to be overwritten as they would affect evaluation semantics in ways that might break the language guarantees.

If that's the case I will refrain from the `&&` overload an use some explicit `and` (global or instance overload). Seems like the better way forward then.
